File: src/main/java/imericxu/zhiheng/mazegen/maze/maze_algos/MazeAlgorithm.java
package imericxu.zhiheng.mazegen.maze.maze_algos;

import imericxu.zhiheng.mazegen.maze.Algorithm;
import imericxu.zhiheng.mazegen.maze.Node;
import imericxu.zhiheng.mazegen.maze.State;

import java.util.Arrays;

public abstract class MazeAlgorithm extends Algorithm
{
	public MazeAlgorithm(Node[] nodes)
	{
		super(nodes, new State[nodes.length]);
		Arrays.fill(states, State.EMPTY);
	}
}

File: src/main/java/imericxu/mazegen/core/maze_algorithms/MazeAlgorithm.kt
package imericxu.mazegen.core.maze_algorithms

import imericxu.mazegen.core.Algorithm
import imericxu.mazegen.core.Node
import imericxu.mazegen.core.State

abstract class MazeAlgorithm(nodes: Array<Node>) : Algorithm(nodes) {
	override val states: Array<State> = Array(nodes.size) { State.EMPTY }
}
